### PR TITLE
[POAE7-2667] Add compression library

### PR DESCRIPTION
**Pull request by xieqi**
### What changes were proposed in this pull request?

Add compression

### Why are the changes needed?

Add compression support

### Does this PR introduce _any_ user-facing change?

No

### How was this patch tested?

Using exist UTs

### Which label does this PR belong to?

infa

### CODE REVIEW

**Comment by winningsix** [comment]
regards to name convention, component/src or scr/component?

**Comment by xieqi** [comment]
> regards to name convention, component/src or scr/component?

seems most is src/component

File: CMakeLists.txt [hunk 1]
```diff
@@ -120,11 +121,15 @@ include_directories(${SUBSTRAITCPP_INCLUDE_DIR})
 #
 # Subdirectories
 #
-include_directories(cider)
-include_directories(cider/include)
-add_subdirectory(cider)
+include_directories(src/cider)
```
Comment (Ma-Jian1) [inline CMakeLists.txt:124]: if we add some other language bind(which doesn't use cmake), then we should change `src` to `cpp` and delete top-level `CMakeLists.txt`

Comment (xieqi) [reply CMakeLists.txt:124]: if add other language, the folder structure likes
```
├── cpp
│   ├── CMakeLists.txt
│   └── src
│       ├── cider
│       ├── cider-velox
│       └── compression
└── java
    ├── cider
    ├── compression
    └── pom.xml

```

Comment (Ma-Jian1) [reply CMakeLists.txt:124]: OK

**Review by Ma-Jian1 — APPROVED** [review]

**Comment by winningsix** [comment]
There're some conflicts. Please help rebase them. @xieqi

**Review by harborn — APPROVED** [review]

**Review by kyotoyx — APPROVED** [review]

**Pull request merged**